packages/commerce/frontend/src/lib/fetcher.ts

```diff
@@ -0,0 +1,35 @@
+import { API_BASE_URL } from './constants.ts'
+import { ErrorResponse, SuccessReponse } from './types.ts'
+
+export class APIError extends Error {
+  public readonly success!: boolean
+  public readonly errors?: Record<string, string>
+
+  constructor(
+    success: boolean,
+    message: string,
+    errors?: Record<string, string>
+  ) {
+    super(message)
+    this.success = success
+    this.errors = errors
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function fetcher<JSON = any>(
+  input: string,
+  options?: RequestInit
+): Promise<SuccessReponse<JSON>> {
+  const response = await fetch(API_BASE_URL + input, {
```

Comment:
Should we stick with `fetch` or use a HTTP client?

Reply:
Well...if we think about the open payments we need to implement in e-commerce, maybe httpclient will be better in the long run?